modules/instance_template/variables.tf

```diff
@@ -171,6 +171,25 @@ variable "shielded_instance_config" {
   }
 }
 
+###########################
+# Confidential Compute VMs
+###########################
+variable "enable_confidential_vm" {
+  default     = false
+  description = "Whether to enable the Confidential VM configuration on the instance. Note that the instance image must support Confidential VMs. See https://cloud.google.com/compute/docs/images"
+}
+
+variable "confidential_instance_config" {
+  description = "Not used unless enable_confidential_vm is true. Confidential VM configuration for the instance."
+  type = object({
+    enable_confidential_compute = bool
```

Comment:
Is this redundant with var.enable_confidential_vm?

Reply:
yeah, it really is since its just one flag at the moment...however, i've left it mirroring the sheilded config in the event additional flags get added into conf.compute (similar to shieldedVMs now)

Reply:
IMO we should stick to just enable_confidential_vm variable for now. Future flags would need a change anyways, so we could add the object (if necessary) later.

Reply:
next push will have just one variable `enable_confidential_vm` which is passed through as

```hcl
  confidential_instance_config {
      enable_confidential_compute = var.enable_confidential_vm
  }
```